[search engine] Add Demonoid search engine

ngosang opened the pull request:
Requested here #2873.

ngosang commented:
@Pireo the problem with the "&" is fixed here.

DoumanAsh commented:
@ngosang Once it will be merge please ping me to delete engine from my repo and unofficial list.

Pireo commented:
Good work guys, now it's with https too, greayìt! Thanks.

Pireo commented:
ngosang, don't you agree about placing torrentreactor in the unofficial list? I mean, it's the only one that finds totally unrelated content (btdigg at least find it in the description if it is not in the title) and we would have less to constantly take care of. 8 is a perfect numer as official.

ngosang commented:
> ngosang, don't you agree about placing torrentreactor in the unofficial list?

Torrentreactor works for me in qBittorrent. In the web http://torrentreactor.com I get unrelated results as you said but in qBittorrent works well. Can you test it a bit more?

Pireo commented:
Exemple:
	  [url=http://img120.imagevenue.com/img.php?image=869585601_Cattura1_122_30lo.jpg][img=http://img120.imagevenue.com/loc30/th_869585601_Cattura1_122_30lo.jpg][/url]

ngosang commented:
I get these results too but with other search word like **thrones** results are good. If @DoumanAsh also thinks it's useless we can move it to unofficial engines.

DoumanAsh commented:
Well i'm not sure if there is point to move it out until it works.
Personally i don't care, but there is no harm to leave it as it is. After all you can opt out the unneeded engine

Pireo commented:
Ok, sorry. I see. I just thought it would be easier for developers too having not too many official engines to take care of, and torrentreactor many times just mess up the search, but if you want to keep it, ok.

sledgehammer999 commented:
@ngosang ping me when this is ready to merge. I'll review and merge.

ngosang commented:
@sledgehammer999 ping.

ngosang commented:
I found a little bug in @DoumanAsh PR. Wait for the commit.

DoumanAsh commented:
@ngosang ping me when you'll make commit. I'm curious to see :)

Pireo commented:
Honestly, I think the previous png was better for BTDigg

ngosang commented:
In the commit **Minor fixes in search engines** I made this changes:
* versions.txt : **removed duplicate extratorrent** and put some order
* btdigg : different version numeration in Python2 & Python 3 versions. I updated the icon also.
* legittorrents : different version numeration in Python2 & Python 3 versions. I updated the name to pair with website.
* kickasstorrents : I updated the name to pair with website and also the URL (the current one works but it's a redirection).

> Honestly, I think the previous png was better for BTDigg

I think we must use the same they have. https://btdigg.org/favicon.ico

@sledgehammer999 ping @DoumanAsh pong

DoumanAsh commented:
```removed duplicate extratorrent and put some order```

Oi oi :)
I see...

Pireo commented:
ngosang, does kickass give you "search aborted" sometimes even if finds a lot of content? Does it take much more time compared to others? With python 2.7.10 it does this to me.

ngosang commented:
> ngosang, does kickass give you "search aborted" sometimes even if finds a lot of content? Does it take much more time compared to others? With python 2.7.10 it does this to me.

Just tested and It's working wery well. Did you tried kat.cr version? https://github.com/ngosang/qBittorrent/blob/demonoid/src/searchengine/nova/engines/kickasstorrents.py

Pireo commented:
I've updated them, but the site displayed is still kickass.to
p.s. the original domani of TPB is working again now

ngosang commented:
> I've updated them, but the site displayed is still kickass.to

Then you didn't update well.
![](http://i.imgur.com/Iw9moNj.png)

Pireo commented:
Maybe different installations caused this, the code is not clean, if it works for you, I'm 100% sure it's ok.

sledgehammer999 commented:
Thank you.
PS: Did you optimize/compress the new pngs ?

ngosang commented:
I was thinking on it but were going to save 20 bytes...

ngosang commented:
@Pireo Can you try to update the plugins with the "Chack for updates" button in qBittorrent?

Pireo commented:
Oh now it works! I did it previously, but with qBT 3.1.12, with 3.2.0 after a re-start now is working, thanks, but it still takes time with kat.cr...

Pireo commented:
I've found legittorrents has the same bug with "&" character and similar

ngosang commented:
> I've found legittorrents has the same bug with "&" character and similar

Will be fixed here #3179.
